graylog2-web-interface/src/views/components/QueryTabs.tsx

```diff
@@ -15,128 +15,241 @@
+      topTabItem.classList.remove(CLASS_HIDDEN);
+      tabItem.classList.add(CLASS_HIDDEN);
+
+      parent.insertBefore(moreBtn, topTabItem);
+
+      adjustTabs();
     }
+  });
+
+  if (!hiddenItems.length) {
+    moreBtn.classList.add(CLASS_HIDDEN);
   }
+};
 
-  render() {
-    const {
-      children,
-      onRemove,
-      onSelect,
-      onTitleChange,
-      queries,
-      selectedQueryId,
-      titles,
-    } = this.props;
-    const queryTitles = titles;
-    const queryTabs = queries.map((id, index) => {
-      const title = queryTitles.get(id, `Page#${index + 1}`);
-      const tabTitle = (
-        <QueryTitle active={id === selectedQueryId}
-                    id={id}
-                    onClose={() => onRemove(id)}
-                    openEditModal={this.openTitleEditModal}
-                    title={title} />
-      );
-
-      return (
-        <Tab eventKey={id}
-             key={id}
-             mountOnEnter
-             title={tabTitle}
-             unmountOnExit>
-          {children}
-        </Tab>
-      );
-    });
-    const newTab = <Tab key="new" eventKey="new" title="+" />;
-    const tabs = [queryTabs, newTab];
-
-    return (
-      <Row style={{ marginBottom: 0 }}>
-        <Col>
-          <StyledQueryTabs activeKey={selectedQueryId}
-                           animation={false}
-                           id="dashboard-pages"
-                           onSelect={onSelect}>
-            {tabs}
-          </StyledQueryTabs>
-          {/*
+const QueryTabs = ({ onRemove, onSelect, onTitleChange, queries, selectedQueryId, titles }:Props) => {
+  const queryTitleEditModal = useRef<QueryTitleEditModal | undefined | null>();
+  const [openedMore, setOpenedMore] = useState<boolean>(false);
+
+  const openTitleEditModal = (activeQueryTitle: string) => {
+    if (queryTitleEditModal) {
+      queryTitleEditModal.current.open(activeQueryTitle);
+    }
+  };
+
+  const queryTabs = (itemType: 'navItem' | 'menuItem') => queries.map((id, index) => {
+    const title = titles.get(id, `Page#${index + 1}`);
+    const tabTitle = (
+      <QueryTitle active={id === selectedQueryId}
+                  id={id}
+                  onClose={() => onRemove(id)}
+                  openEditModal={openTitleEditModal}
+                  title={title} />
+    );
+
+    const output = {
+      navItem: <NavItem eventKey={id} key={id} onClick={() => onSelect(id)}>{tabTitle}</NavItem>,
+      menuItem: <MenuItem eventKey={id} key={id} onClick={() => onSelect(id)}>{tabTitle}</MenuItem>,
+    };
+
+    return output[itemType];
+  });
+
+  const newTab = (
+    <NavItem key="new"
+             eventKey="new"
+             onClick={() => onSelect('new')}
+             className="query-tabs-new">
+      <Icon name="plus" />
+    </NavItem>
+  );
+
+  const dropDownTabs = queryTabs('menuItem');
+
+  useEffect(() => {
+    adjustTabs();
+    window.addEventListener('resize', throttle(adjustTabs, 250));
```

Comment:
A potentially easier solution would also be to use the `SizeMe` component from `react-sizeme` instead, which could potentially also simplify the logic used to generate the tabs. At least we can hand off the size provisioning and handling of window resizes to it.